app/form_objects/forms/payment_link_form.rb

```diff
@@ -0,0 +1,21 @@
+require "uri"
+
+class Forms::PaymentLinkForm < BaseForm
+  attr_accessor :form, :payment_url
+
+  PAY_URL_REGEX = /\Ahttps:\/\/www\.gov\.uk\/payments\/.+\/.+/
```

Comment:
The `.` in this regex will match `/`, so this regex will match `https://www.gov.uk/payments/A/B/C/D/...`. Is that okay?

```
irb(main):003> Forms::PaymentLinkForm::PAY_URL_REGEX.match("https://www.gov.uk/payments/something/something/something/something")
=> #<MatchData "https://www.gov.uk/payments/something/something/something/something">
irb(main):004> Forms::PaymentLinkForm::PAY_URL_REGEX.match("https://www.gov.uk/payments/something")
```

Reply:
We might benefit from using the [uri module](https://docs.ruby-lang.org/en/2.1.0/URI.html) here, so that we can validate bits of the URL individually rather than write one monster regex

Reply:
yeah let's not use regex - Ruby's String.starts_with() AND use URI.parse() https://ruby-doc.org/stdlib-2.5.1/libdoc/uri/rdoc/URI/RFC2396_Parser.html

Reply:
Changes made, @lfdebrux if they do manage to paste something in that doesn't match the link that pay gives them but is still a valid url, it'll be hard for us to spot it. Hopefully their testing will reveal that it's not quite right

Reply:
I also asked Pay if they had a good way of validating and they said what we're doing is legit

Reply:
That sounds good enough, nice one @SamJamCul!